ral-facilities/operationsgateway
# DSEGOG-98 DSEGOG-99 search by predefined and custom relative timeframes

## Description
This adds to the search bar by allowing the user to search by predefined and custom relative timeframes. It includes buttons to search by last 10 mins, last 24 hours and last 7 days. Below this, it includes an input box to enter a user-specified number of minutes, hours or days.

When these buttons are clicked, the date-time values in the date-time pickers are updated accordingly. The timeframe itself is not added as part of the query: it is solely used to change these date-time values.

## Testing instructions
I've done my best to ensure that date-time values can be updated in the date-time pickers themselves and the timeframe component. This required multiple variables in different component states to keep track of things but we need to make sure this is all working before merging this. Please test this as much as possible. I've added comments where I can to help explain what's happening.


## Agile board tracking
Closes DSEGOG-98 DSEGOG-99
